documentation/config_logging_advanced.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: default
+title: "Puppet Server: Advanced Logging Configuration"
+canonical: "/puppetserver/latest/config_logging_advanced.html"
+---
+
+Puppet Server uses the very flexible, very powerful [Logback](http://logback.qos.ch/) library under the hood to handle all of our logging needs.
+
+Among the many interesting and useful capabilities of logback is the ability to configure it to log messages in a JSON format, which makes it easy to send them over to other logging backends such as logstash.
+
+## Configuring Puppet Server for use with Logstash
+
+There are a few steps necessary to get your Puppet Server logging set up for use with logstash.  The first step is to modify your logging configuration so that Puppet Server is logging in a JSON format.  After that, you'll need to configure an external tool to monitor these JSON files and send the data to logstash (or another remote logging system).
+
+### Configuring Puppet Server to log to JSON
+
+There are a handful of considerations you may want to take into account before configuring Puppet Server to log to JSON:
+
+* Do you want to configure Puppet Server to *only* log to JSON?  In lieu of the normal text file logging?  Or do you want to add JSON logging *in addition to* your normal text logging?
+* Do you want to set up JSON logging only for the normal Puppet Server logs (i.e. `puppetserver.log`), or also for the HTTP access logs (i.e. `puppetserver-access.log`)?
+* What kind of log rotation strategy do you want to use for the new JSON log files?
+
+In this section we'll show some examples of setting up your configuration for logging both the normal logs and the HTTP access logs in JSON, *in addition to* leaving the normal text-based log files in place.  We'll also configure logback to do log rotation on these new JSON files.  Hopefully you can modify these examples to suit your own needs, based on your answers to the three questions above.
+
+**NOTE**: Puppet Server's packaging sets up `logrotate` management of the main `puppetserver.log` file automatically, which is why you don't see any of the `logback` support for log rotation configured in our default `logback.xml` file.  However, the Puppet Server packaging does *not* include any `logrotate` integration for these new JSON log files that we'll be setting up, so that is why we will show examples of how to use `logback` to do that.  If you were so inclined, you could omit those portions of these configuration examples, and configure `logrotate` to manage the log rotation for your new JSON files.
+
+#### Adding a JSON version of the "normal" Puppet Server logs
+
+To configure Puppet Server to log its "normal" logs to a second log file, in JSON format, simply add a section like this to your `logback.xml` file:
```

Comment:
Does the location where this is added matter? Can you add it to the beginning of the file, or the end, or anywhere? Is there a part of the file where it _shouldn't_ be added? (I'm assuming it shouldn't be inside another XML element?)